core/src/main/java/org/apache/calcite/sql/SqlCreateProcedure.java

```diff
@@ -105,12 +105,7 @@ public SqlCreateProcedure(SqlParserPos pos, SqlCreateSpecifier createSpecifier,
 
   @Override public void validate(SqlValidator validator,
       SqlValidatorScope scope) {
-    if (statement instanceof SqlScriptingNode
-        || statement instanceof SqlSelect
-        || statement instanceof SqlDelete
-        || statement instanceof SqlInsert
-        || statement instanceof SqlMerge
-        || statement instanceof SqlUpdate) {
+    if (validator.shouldValidateInCreateProcedure(statement)) {
```

Comment:
nit: We might even be able to move even more logic into the shared SqlValidator function.

What if SqlCreateProcedure's validate() method just became:
`validator.validateScriptingStatement(statement, scope);`

And SqlScriptingNode's validate() method became:
`for (SqlNode node : nodes) {
  validator.validateScriptingStatement(node, scope);
}`

And then the if-check and the actual call to validate() could all be handled within SqlValidator's validateScriptingStatement() method.

Reply:
Ah that's true, I like this method better. I'll update and fix the parentheses issue too